### PR TITLE
Migrate documentation from `mkdocs` to Hugo

### LINKED ISSUE #927: Migrate documentation to Hugo
**Is your feature request related to a problem?**
MKDocs has been great for smaller documentation, but as our requirements grow for our website we would prefer to upgrade to a more customisable system.

**Describe the solution you'd like**
Migration of the current user, contributor and reference documentation to a Hugo themed documentation website.

### PR DESCRIPTION
**Pull request by RedbackThomson**
Fixes #927

Description of changes:
Migrates the existing documentation website from the static `mkdocs` site generator to a more customisable Hugo one.
- Uses the [`doks`](https://getdoks.org/) theme as the base
- Uses the AWS brand colours as the accents, defaults to dark theme
- Reworked all of the API reference generator code for Hugo compatibility

Please see the preview for code review, rather than trying to traverse the large number of code changes:

https://redbackthomson.github.io/ack-community/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by RedbackThomson** [comment]
Some things that are still need work:
- The search for the reference pages
- The front page (@jaypipes has ideas!)
- Custom styling for the hint boxes can be cleaned up
- The images need a soft gray background - or to be regenerated with neutral colours

**Comment by RedbackThomson** [comment]
/hold
Need to merge CD before we can merge these changes (so they are deployed upon being merged) 
https://github.com/aws-controllers-k8s/test-infra/pull/104

File: docs/assets/scss/common/_variables.scss
Comment (jaypipes) [inline docs/assets/scss/common/_variables.scss:14]: LOL.

**Review by jaypipes — APPROVED** [review]
No use reviewing all the specific file changes. I'm good with merging this as is. Awesome work, @RedbackThomson!

**Comment by RedbackThomson** [comment]
Need to fix all links to use new relative paths

**Comment by surajkota** [comment]
Overall looks good, few comments before merging

- find a way to keep the website in dark or light mode? I need to switch each page
  - Maybe have light mode as default because inline code blocks `like these` and images are hard to understand in dark mode 
- I feel the Getting started page is more fit as the first page in user docs and Prologue should have an overview/Introduction page as the first page(reach here when you click [get started](https://redbackthomson.github.io/ack-community/)).
  - Can use the content from here https://aws-controllers-k8s.github.io/community/#aws-controllers-for-kubernetes
- Wdyt about renaming Installing a controller to user-docs?
  - that section is installing + configuration + clean up
  - user-docs and contributor-docs seem more intuitive
- there are indexes on both sides of the website which leaves less area of the content itself if looking in a small laptop screen, check if there is a way to collapse and expand those
- can decrease the font size
- nice having API reference as a separate tab

**Comment by RedbackThomson** [comment]
>     * find a way to keep the website in dark or light mode? I need to switch each page       
>       * Maybe have light mode as default because inline code blocks `like these` and images are hard to understand in dark mode
Fixed in my latest push!

>     * I feel the Getting started page is more fit as the first page in user docs and Prologue should have an overview/Introduction page as the first page(reach here when you click [get started](https://redbackthomson.github.io/ack-community/)).
>       * Can use the content from here https://aws-controllers-k8s.github.io/community/#aws-controllers-for-kubernetes
Happy to change content in a different PR. This one is already way too big.

>     * Wdyt about renaming Installing a controller to user-docs?
>       * that section is installing + configuration + clean up
>       * user-docs and contributor-docs seem more intuitive
That is fair, we do more than just install it. "User documentation" is the default case for these docs, though. "Contributor documentation" is the odd one out, everything else should be for the user. 

>     * there are indexes on both sides of the website which leaves less area of the content itself if looking in a small laptop screen, check if there is a way to collapse and expand those
On smaller screens the table of contents should disappear automatically. Decreasing font size should help with this problem, too.

>     * can decrease the font size
Yes, I agree it's too big. Reduced down to 90% of what it was

**Comment by vijtrip2** [comment]
Very Nice!
/approve

**Comment by vijtrip2** [comment]
> /hold
> Need to merge CD before we can merge these changes (so they are deployed upon being merged)
> [aws-controllers-k8s/test-infra#104](https://github.com/aws-controllers-k8s/test-infra/pull/104)

PR is merged now.

**Review by vijtrip2 — APPROVED** [review]

**Review by A-Hilaly — APPROVED** [review]
Neat!! 😍

**Pull request merged**
